packages/commerce-sdk-react/src/hooks/useAuthorizationHeader.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {ApiOptions, ApiMethod} from './types'
+import useAuth from './useAuth'
+
+export const useAuthorizationHeader = <Opts extends ApiOptions, Ret>(
```

Comment:
This replaces `useAuthenticatedClient`, which set the `Authorization` header on the client, with the result of subtly changing all future requests. This method sets the header on the _options_, therefore impacting only the single returned method.